clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -324,6 +325,21 @@ void SYCL::fpga::BackendCompiler::ConstructJob(
       FPGADepFiles.push_back(II);
     else
       CmdArgs.push_back(C.getArgs().MakeArgString(Filename));
+    // Check for any AOCR input, if found use that as the project report name
+    StringRef Ext(llvm::sys::path::extension(Filename));
+    if (Ext.empty())
+      continue;
+    if (getToolChain().LookupTypeForExtension(Ext.drop_front()) ==
+        types::TY_FPGA_AOCR) {
+      // Keep the base of the .aocr file name.  Input file is a temporary,
+      // so we are stripping off the additional naming information for a
+      // cleaner name.
+      SmallString<128> NameBase(Filename.substr(0, Filename.length() - 12));
```

Comment:
I personally find this magic integer constant really difficult to decode. Could it be set into a named constant variable? Or, perhaps, calculated based on the filename + FS specifics?

Reply:
Added some additional doc on how the number is formulated.